pkgs/development/compilers/rust/rustc.nix

```diff
@@ -97,15 +99,18 @@ stdenv.mkDerivation {
   # ps is needed for one of the test cases
   nativeBuildInputs = [ file python2 procps rustPlatform.rust.rustc git ];
   buildInputs = [ ncurses ] ++ targetToolchains
-    ++ stdenv.lib.optional (!forceBundledLLVM) llvmShared;
+    ++ optional (!forceBundledLLVM) llvmShared;
 
   # https://github.com/rust-lang/rust/issues/30181
   # enableParallelBuilding = false; # missing files during linking, occasionally
 
   outputs = [ "out" "doc" ];
   setOutputFlags = false;
 
-  preCheck = "export TZDIR=${tzdata}/share/zoneinfo";
+  preCheck = ''
+    export TZDIR=${tzdata}/share/zoneinfo
+    ${optionalString stdenv.isDarwin "export TMPDIR=/tmp"}
```

Comment:
no it's something along the lines of `/var/folders/ns/m_75wwvs38j6040135v1k9g40000gn/T/`, this cause issues for unix domain sockets created in `$TMPDIR` since the path is longer.